processor/tailsamplingprocessor/internal/sampling/ottl.go

```diff
@@ -0,0 +1,132 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sampling // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor/internal/sampling"
+
+import (
+	"context"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/expr"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspan"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspanevent"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+type ottlStatementFilter struct {
+	sampleSpanExpr      expr.BoolExpr[ottlspan.TransformContext]
+	sampleSpanEventExpr expr.BoolExpr[ottlspanevent.TransformContext]
+	ErrorMode           ottl.ErrorMode
+	logger              *zap.Logger
+}
+
+var _ PolicyEvaluator = (*ottlStatementFilter)(nil)
+
+// evalFuncChain contains eval functions for:
+// 1. span expression
+// 2. span event expression
+var evalFuncChain = []func(ctx context.Context, osf *ottlStatementFilter, span ptrace.Span, scope pcommon.InstrumentationScope, resource pcommon.Resource) (Decision, error){
+	evalSpan,
+	evalSpanEvent,
+}
+
+// NewOTTLStatementFilter looks at the trace data and returns a corresponding SamplingDecision.
+func NewOTTLStatementFilter(logger *zap.Logger, spanStatements, spanEventStatements []string, errMode ottl.ErrorMode) (PolicyEvaluator, error) {
```

Comment:
Can this function take a `component.TelemetrySettings` instead and pass in the processor's instance?

Reply:
```go
func NewOTTLStatementFilter(set component.TelemetrySettings, spanStatements, spanEventStatements []string, errMode ottl.ErrorMode) (PolicyEvaluator, error) {
```

Do you mean (currently) we should pass the logger by a `component.TelemetrySettings`? I am not so sure that the expected signature. But either way, this may need @jpkrohling 's opinion since all available filters are similar to the current format.

Reply:
The current implementation passes a noop logger to the filterottl function which isn't good. In general it's best to pass the component's telemetry settings to OTTL but if we can't do that at least using the logger in the new struct would be required. 

To us telemetry settings not all the other policies would have to change since this New function isn't part of the interface.

Reply:
I see. Thanks for pointing out my mistake. The logger is fixed already. Now I am thinking about how the signature should be and how to organize it in caller function .